operator/internal/central/extensions/reconcile_defaulting.go

```diff
@@ -0,0 +1,131 @@
+package extensions
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/helm-operator-plugins/pkg/extensions"
+	"github.com/pkg/errors"
+	platform "github.com/stackrox/rox/operator/api/v1alpha1"
+	"github.com/stackrox/rox/operator/internal/common/defaulting"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var defaultingFlows = []defaulting.CentralDefaultingFlow{
+	defaulting.CentralScannerV4DefaultingFlow,
+}
+
+// This extension executes "defaulting flows". A Central defaulting flow is of type
+// defaulting.CentralDefaultingFlow, which is essentially a function of type
+//
+//	func(
+//	  logger logr.Logger,
+//	  status *platform.CentralStatus,
+//	  annotations map[string]string,
+//	  spec *platform.CentralSpec,
+//	  defaults *platform.CentralSpec) error
+//
+// A defaulting flow shall
+//   - derive default values based on 'status', 'annotations' and 'spec' and store them in 'defaults'.
+//   - add a new annotation in order to persist current defaulting choices.
+func FeatureDefaultingExtension(client ctrlClient.Client) extensions.ReconcileExtension {
+	return func(ctx context.Context, u *unstructured.Unstructured, _ func(extensions.UpdateStatusFunc), l logr.Logger) error {
+		return reconcileFeatureDefaults(ctx, client, u, l)
+	}
+}
+
+func reconcileFeatureDefaults(ctx context.Context, client ctrlClient.Client, u *unstructured.Unstructured, logger logr.Logger) error {
+	logger = logger.WithName("extension-feature-defaults")
+	if u.GroupVersionKind() != platform.CentralGVK {
+		logger.Error(errUnexpectedGVK, "unable to reconcile central", "expectedGVK", platform.CentralGVK, "actualGVK", u.GroupVersionKind())
+		return errUnexpectedGVK
+	}
+
+	if u.GetDeletionTimestamp() != nil {
+		logger.Info("skipping extension run due to deletionTimestamp being present on Central custom resource")
+		return nil
+	}
+
+	central := platform.Central{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &central); err != nil {
+		return errors.Wrap(err, "converting unstructured object to Central")
+	}
+
+	err := executeDefaultingFlows(ctx, logger, &central, client)
+	if err != nil {
+		return err
+	}
+
+	updatedObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&central)
+	if err != nil {
+		return errors.Wrap(err, "converting Central to unstructured object after extension execution")
+	}
+	u.Object = updatedObj
```

Comment:
Will this work and allow us to avoid a variable?
```suggestion
	u.Object, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&central)
	if err != nil {
		return errors.Wrap(err, "converting Central to unstructured object after extension execution")
	}
```

Reply:
done